### DISCUSSION TITLE
BME688 Bosch API integration with zephyr

### DISCUSSION THREAD

**Discussion by jk0010** [zephyrproject-rtos/zephyr#70384] [Q&A]
Hello,
Has anyone implemented the BME68X Sensor API from Bosch into Zephyr? I know that zephyr already has drivers implemented, I'm looking to implement parallel mode via the api

**Comment by kartben** [comment]
Maybe @pillo79 can chime in, pretty sure he's been looking into it :)

**Comment by dottspina** [comment]
Hi @jk0010 , I had the same kind of issue (mine was especially IAQ), and I just made public my homework on integrating the BME68X Sensor API and the BSEC library with Zephyr: this will encourage me to keep it in a *stable* state (something that should work almost OOTB in case I want to play with my BME680 again), and may help others not to reinvent the wheel, or simply as a source of inspiration to reinvent their own wheel.

This might come close to what you were looking for. The GitHub repository for the Zephyr module (driver, libraries, sample applications) is [bme68x-zephyr](https://github.com/dottspina/bme68x-zephyr).
